lib/src/sql/array.rs

```diff
@@ -121,20 +121,20 @@ impl Array {
 		txn: &Transaction,
 		doc: Option<&Value>,
 	) -> Result<Value, Error> {
-		let mut x = Vec::new();
+		let mut x = Self::with_capacity(self.len());
 		for v in self.iter() {
 			match v.compute(ctx, opt, txn, doc).await {
 				Ok(v) => x.push(v),
 				Err(e) => return Err(e),
 			};
 		}
-		Ok(Value::Array(Array(x)))
+		Ok(Value::Array(x))
 	}
 }
 
 impl fmt::Display for Array {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		write!(f, "[{}]", self.iter().map(|ref v| format!("{}", v)).collect::<Vec<_>>().join(", "))
+		write!(f, "[{}]", self.iter().map(|ref v| v.to_string()).collect::<Vec<_>>().join(", "))
```

Comment:
It should be noted that `ToString::to_string` is implemented by calling `Display::fmt` on an empty string buffer, so this is equivalent (but less codegen)